frontend/test/resolution/testParams.cpp

```diff
@@ -86,10 +86,97 @@ static void test3() {
   assert(qt.param() == ComplexParam::get(&ctx, {3.0, 2.0}));
 }
 
+static void test4() {
+  printf("test4\n");
+  Context ctx;
+  Context* context = &ctx;
+  std::string program = R""""(
+    enum myEnum {
+      blue,
+      red,
+      green
+    }
+
+    proc param myEnum.isBlue() param
+      do return this == myEnum.blue;
+    param x = myEnum.green.isBlue();
+    param y = myEnum.blue.isBlue();
+  )"""";
+
+  auto path = UniqueString::get(context, "test4.chpl");
+  setFileText(context, path, program);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+  assert(vec.size() == 1);
+  const Module* m = vec[0]->toModule();
+  assert(m);
+  assert(m->numStmts() == 4);
+  auto enumDecl = m->stmt(0);
+  assert(enumDecl);
+  auto blueEnum = enumDecl->child(0);
+  auto greenEnum = enumDecl->child(2);
+  assert(blueEnum);
+  assert(greenEnum);
+  auto xStmt = m->stmt(2);
+  auto yStmt = m->stmt(3);
+  assert(xStmt);
+  assert(yStmt);
+  const ResolutionResultByPostorderID& rr = resolveModule(context, m->id());
+  const auto& resolvedXExpr = rr.byAst(xStmt);
+  const auto& resolvedYExpr = rr.byAst(yStmt);
+  assert(resolvedXExpr.type().kind() == QualifiedType::Kind::PARAM);
+  assert(resolvedYExpr.type().kind() == QualifiedType::Kind::PARAM);
+  assert(resolvedXExpr.type().param()->isBoolParam());
```

Comment:
Should `QualifiedType::isParamTrue()` do all of this checking for you?

Reply:
yep, replaced!